### DISCUSSION TITLE
Sentinel - get all data bands? (10m or 20m).

### DISCUSSION THREAD

**Discussion by RichardScottOZ** [developmentseed/titiler#196] [Q&A]
Is your mosaic tiler idea only supposed to grab 3 bands at a time? e.g. could hack to return say, the 4 10m Sentinel bands? Thanks!

My use case to modify this is to grab data mosaics, the visualisation doesn't matter, just the original DNs.

Getting a web map application that can visualise any 3 band look is certainly a nice bonus though.

**Comment by kylebarron** [comment]
You can grab as many bands at a time as you want, but the output format needs to support it. E.g. if you give 4 bands to a PNG it might make the last one alpha as default. But a format like "numpy tile" can output as many bands as you want. Just pass four items to `indexes` (edit: maybe it's `bands`, not `indexes`, I can't remember)

**Reply by RichardScottOZ** [reply]
Thanks Kyle.

**Comment by vincentsarago** [comment]
@RichardScottOZ, following @kylebarron excellent response

in the STAC reader we use `bands` to pass the list of bands we want to fetch.

If you want bands B01, B02 you'll need to pass `bands="B01,B02"`

rio-tiler-pds docs: https://cogeotiff.github.io/rio-tiler-pds/usage/sentinel/#cog-only-l2a-available-for-now
> supported bands: B01, B02, B03, B04, B05, B06, B07, B08, B09, B11, B12, B8A

titiler-pds uses `bands`: https://github.com/developmentseed/titiler-pds/blob/master/app/dependencies.py#L46-L48
while the titiler STAC endpoints uses `assets` to specify the assets within the STAC item.

as @kylebarron mentioned, to return more than 3 bands in a native datatype (uint16), you'll need to use the NumpyTile or Tiff format. Docs: https://developmentseed.org/titiler/concepts/output_format/

**Reply by RichardScottOZ** [reply]
Thanks very much Vincent, I will give that a go next week.

**Reply by RichardScottOZ** [reply]
So basically something along these lines?

`tilejunk` = requests.get(f"{titiler_endpoint}/scenes/sentinel/tiles/{tile.z}/{tile.x}/{tile.y}.npy?sceneid={sceneid[4]}&bands=B04,B03,B02,B08")
`print(tilejunk)`

If I understand it correctly, what you have created in titiler-pds is a version of:-

devseed.com/titiler/concepts/tiler_factories/ ?

**Reply by RichardScottOZ** [reply]
The .tif version - is that just returning an 8byte 3 band picture product?  Or can you use any number of bands?  Is there a recent example/function somewhere where you take .tif output, get the appropriate geospatial info write it out?
